### ISSUE TITLE
Possible to use with Expo dev client?

### ISSUE THREAD

**Issue by UchennaOkafor** [mybigday/whisper.rn#61]
This package seems amazing, and being able to transcribe directly on the device is game changing, however I ran into many issues  using it with Expo. I want to remain within the dev-client and don't want to run `npx expo prebuild` because it creates the ios and android files.

Is it possible to make this package compatible with Expo dev client?

**Comment by jhen0409** [comment]
I'm not very familiar with Expo so I quickly read the docs, I guess we can support it by create [Development builds](https://docs.expo.dev/develop/development-builds/introduction). Any PRs are welcome.

**Comment by UchennaOkafor** [comment]
@jhen0409 

It looks like it works with dev client perfectly fine without any need for modifications. Only problem I am having is loading the model. I followed the example you posted in the ReadMe and also in
https://github.com/mybigday/whisper.rn/blob/main/example/src/App.js

but not been able to load the model.

If I create a blank expo dev client repo and replicate the problem, would you be open to taking a look? Thanks

**Comment by UchennaOkafor** [comment]
@jhen0409 
I'd even pay if you can help me getting it to work on expo dev client

**Comment by jhen0409** [comment]
I'll not take any payment, this project is driven by the company.

> 
> It looks like it works with dev client perfectly fine without any need for modifications. Only problem I am having is loading the model. I followed the example you posted in the ReadMe and also in
> 
> https://github.com/mybigday/whisper.rn/blob/main/example/src/App.js
> 
> but not been able to load the model.
> 

Is the problem is load model from assets? (I was found your another comment from another issue)

In general, React Native provided a way to bundle the assets, we can use `require` and `Image.resolveAssetSource`. It resolves a URL in debug mode (provided by packager) and bundle asset path in release mode. Unfortunately this package is not directly works for URL currently, but it probably can easy to support.

> 
> If I create a blank expo dev client repo and replicate the problem, would you be open to taking a look? Thanks

I can take a look for that.

**Comment by UchennaOkafor** [comment]
> I'll not take any payment, this project is driven by the company.
> 
> > It looks like it works with dev client perfectly fine without any need for modifications. Only problem I am having is loading the model. I followed the example you posted in the ReadMe and also in
> > https://github.com/mybigday/whisper.rn/blob/main/example/src/App.js
> > but not been able to load the model.
> 
> Is the problem load model from assets? (I was found your comment from another issue)
> 
> In general, React Native provided a way to bundle the assets, we can use `require` and `Image.resolveAssetSource`. It resolves a URL in debug mode (provided by packager) and bundle asset path in release mode. Unfortunately this package is not directly works for URL currently, but it probably can easy to support.
> 
> > If I create a blank expo dev client repo and replicate the problem, would you be open to taking a look? Thanks
> 
> I can take a look for that.

Thank you so much, I have created an Expo SDK 48 project with the Whsiper.RN module installed.
https://github.com/UchennaOkafor/whisper-rn-expo-dev-client

In the repository, there is a /builds folder, and that's where the iOS and Android app files are (I have taken the time to build the dev client for you). 

1) For the iOS file, you need to unzip it and drag it to your emulator to open to install the dev client.
2) For the Android apk, you need to just drag it to your emulator to install the dev client
3) If you want to build the dev client yourself then you can run `yarn build-dev-client-android` and `yarn build-dev-client-ios`
4) Then just run `yarn android` or `yarn ios` to start.

Once you're there the UI should look like this.

![Simulator Screenshot - iPhone 14 Pro - 2023-07-11 at 22 19 07](https://github.com/mybigday/whisper.rn/assets/22923123/5618fccf-0265-4bcd-827d-955bd39b3cb8)

So you can see that click Initialize from Assets and Initialize Model from URL doesn't work.

I'd honestly appreciate any help because I would really like to use this module within my app, it would make all the difference. Appreciate the help, you're a hero <3

**Comment by jhen0409** [comment]
Thanks for the great example & description, it help me to understand the problem quickly!

After #81 it is supported load local asset. I tested to build local expo dev-client with `eas-cli`, please see https://github.com/UchennaOkafor/whisper-rn-expo-dev-client/pull/1.

**Comment by UchennaOkafor** [comment]
Thank you. I have tested this and it works well, I am able to load from assets and load from path. You have been such a great help, I really appreciate the time you spent working on this. This is a great library and I hope it will become the new go-to library for text to speech in react native.
